bin/stt_test/stt_providers/ISTTBase.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Literal, Protocol, TypedDict
+import speech_recognition as sr  # type: ignore
+import time
+from stt_test.utils.SentenceChecker import SentenceChecker
+from stt_test.utils.test_data import get_audio_with_transcription
+import inspect
+
+
+class STTResult(TypedDict):
+    stt_trans: str
+    trans: str
+    similarity: float
+    execution_time: float
+
+
+class ISSTBase(Protocol):
+
+    MODEL_SIZES: List[str]
+    _sc: SentenceChecker
+    _recognizer: sr.Recognizer
+
+    def _init(self):
+        self._sc = SentenceChecker()
+        self._recognizer = sr.Recognizer()
+
+    def _before_all(self, model_size: str):
+        """Method for preparing models"""
```

Comment:
So how about calling it `prepare_models`?

Reply:
I thought about it, but I decided that it would be safer to create a separate method for this, because one day it may turn out that you will need to do something more than just prepare a model. Let me know what you think

Reply:
So how about removing it and using _prepare_models inside of the _init function?

Reply:
In case of any additional steps we can also call them here.